### DISCUSSION TITLE
Implementation for Let's Encrypt HTTPS (SSL/TLS) for OrchardCore?

### DISCUSSION THREAD

**Discussion by tysongibby** [OrchardCMS/OrchardCore#8645] [Ideas]
If someone else has already started an implementation of Let's Encrypt for OrchardCore, I would like to assist. If no one has started one, I am looking into doing it.

"Let’s Encrypt is a free, automated, and open certificate authority (CA), run for the public’s benefit. It is a service provided by the Internet Security Research Group (ISRG).
We give people the digital certificates they need in order to enable HTTPS (SSL/TLS) for websites, for free, in the most user-friendly way we can. We do this because we want to create a more secure and privacy-respecting Web."  src: [Let's Encrypt - About](https://letsencrypt.org/about/)

**Comment by tysongibby** [comment]
I am currently considering the PKISharp project for this idea. [PKISharp](https://github.com/PKISharp/acmesharpcore)
It is an opensource project here on GitHub with an MIT license that uses the latest version of the Let's Encrypt ACME API. The project is also based in .NET Standard.

**Reply by jeffolmstead** [reply]
@tysongibby there was the concept of a module a while back. However, a lot of us host on Azure and they now have SSL encryption (free) via a service like Let's Encrypt (possibly is Let's Encrypt) so the module was no longer pursued. There is the initial thread here: https://github.com/OrchardCMS/OrchardCore/issues/2607 and actual module work here (along with reason it was canceled): https://github.com/OrchardCMS/OrchardCore/pull/3745

**Reply by tysongibby** [reply]
Thanks Jeff @jeffolmstead ! Makes sense. I did notice that Azure had that feature. It sounds like a very nice one. CertBot is also very easy to install and easy use with an any Apache/Linux server so it works great with .NETCore Kestrel server, but it is not integrated into OrchardCore.  I appreciate the links to the initial thread and work.
